src/router/index.ts

```diff
@@ -141,12 +141,14 @@ export function pathToRoute(path: string | null): Route {
         return {
           type: "faucet",
           params: {
-            activeView: "withdraw",
-            amount: url.searchParams.get("amount"),
+            activeView: {
+              type: "withdraw",
+              params: { amount: url.searchParams.get("amount") ?? "0" },
```

Comment:
I understand this change, but I'd prefer to parse it to `null` and handle it in the `FaucetWithdraw` component.
Since a missing search query means that nothing was passed (not even 0)

Reply:
Before you weren't handling it at all thought:
```
{#if activeView === "form"}
  <Index {wallet} />
{:else if activeView === "withdraw" && amount}
  <Withdraw {wallet} {amount} />
{/if}
```

You didn't render anything.

<img width="1840" alt="Screenshot 2022-10-27 at 15 20 57" src="https://user-images.githubusercontent.com/158411/198295782-01650537-4e9f-4371-9d18-44f9abc27508.png">

Reply:
If we hit the withdraw endpoint manually _with_ an amount, it also doesn't work, because it relies on the form step to do some set-up beforehand. So I think it's out of scope to fix this for the routing PR.

http://127.0.0.1:3000/faucet/withdraw?amount=10

<img width="1840" alt="Screenshot 2022-10-27 at 15 26 27" src="https://user-images.githubusercontent.com/158411/198297114-421ebeec-baf4-4034-8ba6-146b24872ad5.png">

Reply:
Did it like this: 2f6173fd367783d38dd8d12568cc9ac36a76abe7.